lms/services/digest.py

```diff
@@ -86,16 +76,18 @@ def send_instructor_email_digests(
                 continue
 
             try:
-                self._mailchimp_service.send_template(
-                    "instructor-email-digest",
-                    self._sender,
-                    recipient=EmailRecipient(to_email, unified_user.display_name),
+                send_template.delay(
+                    template_name="instructor-email-digest",
+                    sender=asdict(self._sender),
+                    recipient=asdict(
+                        EmailRecipient(to_email, unified_user.display_name)
+                    ),
                     template_vars=digest,
                     unsubscribe_url=self._email_unsubscribe_service.unsubscribe_url(
                         unified_user.h_userid, EmailUnsubscribe.Tag.INSTRUCTOR_DIGEST
                     ),
                 )
-            except MailchimpError as err:
+            except Exception as err:  # pylint:disable=broad-exception-caught
```

Comment:
Just like we were previously catching any exceptions from the Mailchimp API and continuing on to the next user in the `for` loop, we now need to catch any exceptions from Celery/from trying to publish a message to RabbitMQ and continue. We don't want a random Celery/RabbitMQ error for one user in the batch to cause the `for` loop to exit and the rest of the batch to be skipped.

The [Celery docs for `apply_async()`](https://docs.celeryq.dev/en/stable/reference/celery.app.task.html?highlight=apply_async#celery.app.task.Task.apply_async) (which `delay()` is a wrapper for) say that `kombu.exceptions.OperationalError` is raised if there are problems with RabbitMQ. So we may be able to catch just that instead of the broad exception. But I think the broad exception is safer: I don't know for sure that Celery can't raise other types of exceptions here.

Reply:
Yeah this seems one of the use cases for the broad `Exception` catching :+1: